text/0000-grid-primitives.md

```diff
@@ -0,0 +1,712 @@
+    lat_long_value=LatLong(
+        latitude=44977753,
+        longitude=-93265015)
+)
+```
+
+Due to the use of protobuf, the default values for `LatLong` would be `(0, 0)`.
+While this is a valid lat/long, it could be used to indicate an error, depending
+on the choice of the smart-contract implementer.
+
+### Datetime
+
+A datetime value would be represented as follows:
+```
+PropertyDefinition(
+    name='created_at',
+    data_type=PropertyDefinition.DataType.DATETIME,
+    required=True
+)
+```
+
+A datetime instance for this definition would be as follows:
+```
+PropertyValue(
+    name='created_at',
+    data_type-PropertyDefinition.DataType.DATETIME,
+    datetime_value='2019-05-31T14:53:18+0000'
+)
+```
+
+Due to the use of protobuf, the default value for `datetime_value` is an empty
+string, and therefore an invalid value.  This field must be set for the
+`DATETIME` data type.
+
+## Schema Example
+
+A complete object representation can be built from the property definition
+messages, and instances can be represented by constructing items with the
+property value messages.  
+
+Suppose there is a requirement to store different types of light bulbs as part
+of an application. A lightbulb may consist of the properties size, bulb type,
+energy rating, and color.
+
+We could define a Lightbulb schema as follows:
+
+```
+Schema(
+    name="Lightbulb",
+    description="Example Lightbulb schema",
+    owner = "philips001"
+    properties=[
+        PropertyDefinition(
+            name="size",
+            data_type=PropertyDefinition.DataType.NUMBER,
+            description="Lightbulb radius, in millimeters",
+            number_exponent=0,
+            required=True
+        ),
+        PropertyDefinition(
+            name="bulb_type",
+            data_type=PropertyDefinition.DataType.ENUM,
+            enum_options=["filament", "CF", "LED"],
+            required=True
+        ),
+        PropertyDefinition(
+            name="energy_rating",
+            data_type=PropertyDefinition.DataType.NUMBER,
+            number_exponent=-2,
+            description="EnergyStar energy rating (percent)",
+        )
+        PropertyDefinition(
+            name="color",
+            data_type=PropertyDefinition.DataType.STRUCT,
+            description="A named RGB Color value",
+            struct_properties=[
+                PropertyDefinition(
+                    name='name',
+                    data_type=PropertyDefinition.DataType.STRING,
+                ),
+                PropertyDefinition(
+                    name='rgb_hex',
+                    data_type=PropertyDefinition.DataType.STRING,
+                )])])
+```
+
+Note: This example looks very similar to defining a struct property, but the
+fields in a schema may be optional.
+
+We can define a data structure that uses this schema for validation as follows:
+
+```
+message Lightbulb {
+    string id = 1;
+    string production_org = 2;
```

Comment:
`id` and `production_org` are not valid.  Should update this example.

Reply:
How so?  It is merely an example of how the primitives might be used, not an actual implementation.

Reply:
I'm maybe missing something - wouldn't be first time - but the schema example and the message proto don't match. At a minimum it's maybe just unnecessarily confusing to the reader

Reply:
The schema is describing the possible values that would be part of the `properties` field of this protobuf message.  I think I see what you mean about the connection between the two not being clear.

Reply:
Updated the surrounding text with (hopefully) a clearer indication of the relationship between the schema and the instance record.

Reply:
@davececchi Added an example of a lightbulb instance, as well as reworked the example text.